workers/nlp/annotate/annotate.py

```diff
@@ -1,64 +1,35 @@
-class NoTextProvidedException(Exception):
-    pass
+import uuid
 
 
-class InvalidNlpOperationException(Exception):
+class NoTextProvidedException(Exception):
```

Comment:
Warum brauchen wir hier eine Custom Exeption? Reicht für nicht vorhandene Parameter nicht ein TypeError oder ValueError?
Das soll jetzt keine Aufforderung sein, ich bin noch unsicher, wie man das am besten macht. Aber irgenwie erschließt sich der Sinn nicht so recht, Exceptions zu definieren für Dinge die es schon gibt.

Reply:
Hmm, die wird ja bei emptystrings geworfen, damit sich der Analyzer nicht initialisieren muss..
Nen emptystring würde aber keinen TypeError oder ValueError darstellen, theoretisch ginge alles damit. Ist nur nicht sinnvoll.
Sicher bin ich mir aber auch nicht, können wir gern drüber reden

Reply:
Meines Erachtens ist die Exception an der Stelle auch overengineered. Die Methode `_validate_text()` kann man sich aus mehreren Gründen sparen glaube ich:

1) Behauptet der Methodenname etwas, dass die Implementation nicht hält. Sie müsste eigentlich `_assert_text_not_empty()` heißen. `assert` würde darauf hinweisen, dass eine Exception geworfen wird und `text_not_empty` würde näher spezifizieren was genau validiert wird.
2) Wäre der Code lesbarer, wenn man in `annotate` prüft, ob der Text leer ist. Dann müsste man nicht erst im docstring nachlesen, dass die Methode dazu gedacht erst vor dem Initialisieren des Analyzers aufgerufen zu werden. Der Test würde einfach davor stattfinden.
3) Finde ich, dass leerer Text kein exceptional state ist. Es kann ja durchaus vorkommen, dass in einem Buch eine leere Seite vorkommt. Die hat dann eben einfach keine Annotations.

Vorschlag:

```python
def annotate(text, lang=None):
    if not text:
        return {};
    text_analyzer = _init_text_analyzer(text)
    if lang:
        text_analyzer.lang = lang

    return _add_metadata(text_analyzer, _full_ner(text_analyzer))
```